services/app-web/src/pages/App/AppBody.test.tsx

```diff
@@ -72,6 +73,46 @@ describe('App Body and routes', () => {
         })
     })
 
+    it('displays maintenance banner when flag is on', async () => {
+        ldUseClientSpy({ 'site-under-maintenance-banner': 'UNSCHEDULED' })
+        renderWithProviders(<AppBody authMode={'AWS_COGNITO'} />, {
+            apolloProvider: {
+                mocks: [
+                    fetchCurrentUserMock({ statusCode: 200 }),
+                    indexHealthPlanPackagesMockSuccess(),
+                ],
+            },
+        })
+        expect(
+            await screen.findByRole('heading', { name: 'Site unavailable' })
+        ).toBeInTheDocument()
+        expect(
+            await screen.findByText(
+                /MC-Review is currently unavailable due to technical issues/
+            )
+        ).toBeInTheDocument()
+    })
+
+    it('does not display maintenance banner when flag is off', async () => {
+        ldUseClientSpy({})
+        renderWithProviders(<AppBody authMode={'AWS_COGNITO'} />, {
```

Comment:
> I had to call the useClientSpy with {} even when I didn’t want to override the default value, let’s talk about that I think it would be best if you didn’t have to swizzle the method if you just want the defaults.

I think the `jest.resetAllMocks()` at the top level `describe` does not apply to each test in the `describe` block for 'Environment specific banner'

```
describe('App Body and routes', () => {
    afterEach(() => {
        jest.resetAllMocks()
    })
```

Adding a `jest.resetAllMocks()` in the `beforeEach()` should clear out the `useClientSpy` mock that was called in the previous test.

```
    describe('Environment specific banner', () => {
        const OLD_ENV = process.env
        beforeEach(() => {
            jest.resetModules() // Most important - clears the cache
            process.env = { ...OLD_ENV } // Make a copy
        })
```

Reply:
@macrael I ran the tests without calling `useClientSpy({})` and it still passed. Maybe we can pair and you could show me what is going on.

Reply:
Let’s pair on this, what I’m seeing is that if you don’t call ldUseClientSpy anywhere in your test, all of our flag calls just return undefined because the launch darkly client isn’t setup.